llvm/lib/Transforms/Utils/ScalarEvolutionExpander.cpp

```diff
@@ -1293,7 +1293,10 @@ Value *SCEVExpander::visitTruncateExpr(const SCEVTruncateExpr *S) {
 
 Value *SCEVExpander::visitZeroExtendExpr(const SCEVZeroExtendExpr *S) {
   Value *V = expand(S->getOperand());
-  return Builder.CreateZExt(V, S->getType());
+  auto *Res = Builder.CreateZExt(V, S->getType());
```

Comment:
We should add a IsNNeg bool param to IRBuilder::CreateZExt.

Reply:
Agreed.   Once we have a couple examples, a NFC to restructure the API might be called for.

My main point of uncertainty is what we're going to want for the ZExtOrTrunc variants.  Still waiting to see what actual idiomatic usage looks like.